src/types.ts

```diff
@@ -116,9 +116,9 @@ export interface EditorProps<TRow, TSummaryRow = unknown> extends SharedEditorPr
 
 export interface HeaderRendererProps<TRow, TSummaryRow = unknown> {
   column: CalculatedColumn<TRow, TSummaryRow>;
-  sortColumn: string | undefined | null;
-  sortDirection: SortDirection | undefined | null;
-  onSort: ((columnKey: string, direction: SortDirection) => void) | undefined | null;
+  sortDirection?: SortDirection | undefined | null;
+  priority?: number;
+  onSort?: (columnKey: string, direction: SortDirection, ctrlClick: boolean) => void;
```

Comment:
- in HeaderCell, we should always pass `sortDirection`/`priority`/`onSort` so they're not really optional (`?:`)
- `sortDirection` is never `null`
- `onSort` is never `undefined`
```suggestion
  sortDirection: SortDirection | undefined;
  priority: number | undefined;
  onSort: (columnKey: string, direction: SortDirection, ctrlClick: boolean) => void;
```